### PR TITLE
docs: Add InputDate docs

### PR DESCRIPTION
**Pull request by MichaelParadis**
### Added

- Added InputDate docs

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/components/InputDate/Mobile.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    viewMode: "story",
+    previewTabs: { code: { hidden: false } },
+    viewport: { defaultViewport: "mobile1" },
+    showNativeOnWebDisclaimer: true,
+  },
+} as ComponentMeta<typeof InputDate>;
+
+const BasicTemplate: ComponentStory<typeof InputDate> = args => {
+  const [date, setDate] = useState<Date | undefined>(new Date("11/11/2011"));
+  const props = omit({ ...args }, "name", "validations", "defaultValue");
```
[the remark this answers is not in the document]
Comment (MichaelParadis) [reply docs/components/InputDate/Mobile.stories.tsx:19]: Dealing with Typescript issues relating to FormControlled and DevControlled props in the XOR

[the remark this answers is not in the document]
Comment (MichaelParadis) [reply docs/components/InputDate/Mobile.stories.tsx:19]: Also I am making some small tweaks I missed before

File: docs/components/InputDate/InputDate.stories.mdx [hunk 2]
```diff
@@ -65,6 +26,24 @@ input usage, see [Related Components](#related-components).
 
 Accessibility concerns for the InputDate are captured by...
```
Comment (taylorvnoj) [inline docs/components/InputDate/InputDate.stories.mdx:27]:
```suggestion
Accessibility concerns for the InputDate are captured by:
```

File: docs/components/InputDate/InputDate.stories.mdx [hunk 3]
```diff
@@ -1,56 +1,17 @@
...
-        value={date}
-        onChange={setDate}
-      />
-    );
-  }}
-</Canvas>
-
 ## Design & usage guidelines
 
 The `<InputDate>` should be used in most cases where the user needs to select a
```
Comment (taylorvnoj) [inline docs/components/InputDate/InputDate.stories.mdx:17]:
```suggestion
The InputDate should be used in most cases where the user needs to select a
```

File: docs/components/InputDate/InputDate.stories.mdx [hunk 4]
```diff
@@ -89,7 +68,7 @@ group.
 
 ## Related components
```
Comment (taylorvnoj) [inline docs/components/InputDate/InputDate.stories.mdx:69]: Should we follow the order of sections as laid out in the ticket?  Accessibility would come last...
```
Design & usage guidelines

Content guidelines

Related components

Related patterns

Related resources

Platform considerations

Accessibility
```

File: docs/components/InputDate/InputDate.stories.mdx [hunk 5]
```diff
@@ -1,56 +1,17 @@
...
+import { Canvas, Meta } from "@storybook/addon-docs";
 import { InputDate } from "@jobber/components/InputDate";
 
-<Meta title="Components/Forms and Inputs/InputDate" component={InputDate} />
+<Meta title="Components/Forms and Inputs/InputDate/Docs" />
 
 # Input Date
 
 InputDate provides a multi-modal way for the user to select a date, using either
 of:
```
Comment (taylorvnoj) [inline docs/components/InputDate/InputDate.stories.mdx:9]:
```suggestion
InputDate provides a multi-modal way for the user to select a date, using either:
```

File: docs/components/InputDate/Mobile.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import omit from "lodash/omit";
+import { InputDate } from "@jobber/components-native";
+
+export default {
+  title: "Components/Forms and Inputs/InputDate/Mobile",
+  component: InputDate,
+  parameters: {
+    viewMode: "story",
+    previewTabs: { code: { hidden: false } },
```
Comment (taylorvnoj) [inline docs/components/InputDate/Mobile.stories.tsx:11]: Code Tab doesn't work (pretty sure Darryl has a PR open for me to review which fixes this).  Even still, it's import is incorrect (not pulling from coponents-native)

Screenshot 2023-07-19 at 10 31 12 AM

File: docs/components/InputDate/Web.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+export const Basic = BasicTemplate.bind({});
+Basic.args = {
+  placeholder: "Start Date",
+};
+
+const MinMaxTemplate: ComponentStory<typeof InputDate> = args => {
+  const [date, setDate] = useState(new Date("11/11/2011"));
+  return <InputDate {...args} value={date} onChange={setDate} />;
+};
```
Comment (taylorvnoj) [inline docs/components/InputDate/Web.stories.tsx:27]: This template is the exact same as the basic one. You only need the one.  Then in the below export you can just call `BasicTemplate.bind`

**Review by taylorvnoj — CHANGES_REQUESTED** [review]

**Pull request merged**